### PR TITLE
Fixes #12702: Method copyResourceFile is quite inefficient

### PR DESCRIPTION
**Pull request by ncharles**
https://www.rudder-project.org/redmine/issues/12702

### CODE REVIEW

File: rudder-core/src/main/scala/com/normation/rudder/services/policies/write/PromiseWriteDataStructures.scala [hunk 1]
```diff
@@ -122,3 +122,12 @@ case class TechniqueTemplateCopyInfo(
...
   override def toString() = s"Promise template ${id.name}; destination ${destination}"
 }
+
+
+case class TechniqueResourceCopyInfo(
+    id         : TechniqueResourceId
+  , destination: String
+  , content    : String //template resource as a file
+) extends HashcodeCaching {
+  override def toString() = s"Promise resource ${id.name}; destination ${destination}"
```
Comment (fanf) [inline rudder-core/src/main/scala/com/normation/rudder/services/policies/write/PromiseWriteDataStructures.scala]: Can you use something else then "promise" here? Technique resource, or policy resource.

**Comment by fanf** [comment]
Appart for the little comment, this is a very nice pr! 

Oh: could you try to align "<-" too?

**Comment by ncharles** [comment]
Commit modified

File: rudder-core/src/main/scala/com/normation/rudder/services/policies/write/PromiseWriteDataStructures.scala (hunk 1 above)
Comment (ncharles) [reply rudder-core/src/main/scala/com/normation/rudder/services/policies/write/PromiseWriteDataStructures.scala]: ok, renaming previous message also then

**Comment by ncharles** [comment]
Commit modified

**Comment by ncharles** [comment]
Commit modified

**Comment by ncharles** [comment]
OK, merging this PR

**Comment by ncharles** [comment]
OK, merging this PR

**Comment by ncharles** [comment]
OK, merging this PR

**Pull request merged**